tests/test_solution.py

```diff
@@ -576,3 +579,59 @@ def test_serialization(s1, s2, tmpdir):
     # also should point to different Store instances
     # TODO currently this test will fail due to a bug in maggma's __eq__
     # assert s2_new.database != s2.database
+
+
+def test_valid_preset():
```

Comment:
I'd prefer to structure the tests such that there is one test per corresponding method. So in this case, could you please consolidate `test_valid_preset`, `test_invalid_preset`, and `test_correct_data_loading` into a single `test_from_preset` function?